MekHQ/src/mekhq/campaign/personnel/skills/SkillType.java

```diff
@@ -0,0 +1,1933 @@
+
+            for (int x = 0; x < nodeList.getLength(); x++) {
+                Node wn2 = nodeList.item(x);
+                if (wn2.getNodeName().equalsIgnoreCase("name")) {
+                    // skillType.name = wn2.getTextContent();
+
+                    //Start <50.01 compatibility handler.
+                    // The above code can be uncommented once these handlers have been removed
+                    skillType.name = switch (wn2.getTextContent().toLowerCase()) {
+                        case "piloting/mech" -> "Piloting/Mek";
+                        case "gunnery/mech" -> "Gunnery/Mek";
+                        case "gunnery/battlesuit" -> "Gunnery/BattleArmor";
+                        case "gunnery/protomech" -> "Gunnery/ProtoMek";
+                        case "anti-mech" -> "Anti-Mek";
+                        case "tech/mech" -> "Tech/Mek";
+                        case "tech/ba" -> "Tech/BattleArmor";
+                        case "medtech" -> "MedTech";
+                        default -> wn2.getTextContent();
+                    };
+                    //End <50.01 compatibility handler
+                } else if (wn2.getNodeName().equalsIgnoreCase("target")) {
+                    skillType.target = Integer.parseInt(wn2.getTextContent());
+                } else if (wn2.getNodeName().equalsIgnoreCase("greenLvl")) {
+                    skillType.greenLvl = Integer.parseInt(wn2.getTextContent());
+                } else if (wn2.getNodeName().equalsIgnoreCase("regLvl")) {
+                    skillType.regLvl = Integer.parseInt(wn2.getTextContent());
+                } else if (wn2.getNodeName().equalsIgnoreCase("vetLvl")) {
+                    skillType.vetLvl = Integer.parseInt(wn2.getTextContent());
+                } else if (wn2.getNodeName().equalsIgnoreCase("eliteLvl")) {
+                    skillType.eliteLvl = Integer.parseInt(wn2.getTextContent());
+                } else if (wn2.getNodeName().equalsIgnoreCase("countUp")) {
+                    skillType.countUp = Boolean.parseBoolean(wn2.getTextContent().trim());
+                } else if (wn2.getNodeName().equalsIgnoreCase("subType")) {
+                    skillType.subType = SkillSubType.fromString(wn2.getTextContent().trim());
+                } else if (wn2.getNodeName().equalsIgnoreCase("firstAttribute")) {
+                    skillType.firstAttribute = SkillAttribute.fromString(wn2.getTextContent().trim());
+                } else if (wn2.getNodeName().equalsIgnoreCase("secondAttribute")) {
+                    skillType.secondAttribute = SkillAttribute.fromString(wn2.getTextContent().trim());
+                } else if (wn2.getNodeName().equalsIgnoreCase("costs")) {
+                    String[] values = wn2.getTextContent().split(",");
+                    for (int i = 0; i < values.length; i++) {
+                        skillType.costs[i] = Integer.parseInt(values[i]);
+                    }
```

Comment:
MathUtility prasing.... If you need more than the ones currently there, just add a format for the type.